sdk/common/src/main/java/com/mesosphere/sdk/offer/taskdata/LabelReader.java

```diff
@@ -0,0 +1,83 @@
+package com.mesosphere.sdk.offer.taskdata;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.mesos.Protos.HealthCheck;
+import org.apache.mesos.Protos.Labels;
+import org.apache.mesos.Protos.TaskInfo;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.mesosphere.sdk.offer.TaskException;
+
+/**
+ * Implements common logic for read access to a task's Labels. Any access to component-specific values is provided by
+ * ExecutorLabelReader and SchedulerLabelReader.
+ */
+public class LabelReader {
+
+    private final String taskName;
+    private final Map<String, String> labels;
+
+    /**
+     * Creates a new instance for reading the provided task's labels.
+     */
+    protected LabelReader(TaskInfo taskInfo) {
+        this(taskInfo.getName(), taskInfo.getLabels());
+    }
+
+    /**
+     * Creates a new instance for reading the provided task builder's labels.
+     */
+    protected LabelReader(TaskInfo.Builder taskInfoBuilder) {
+        this(taskInfoBuilder.getName(), taskInfoBuilder.getLabels());
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param taskName the name of the task, only used in error messages if there are problems with the label content
+     * @param labels the labels to be read from
+     */
+    private LabelReader(String taskName, Labels labels) {
+        this.taskName = taskName;
+        this.labels = LabelUtils.toMap(labels);
+    }
+
+    /**
+     * Returns the readiness check to be run by the Executor on task startup.
+     */
+    public Optional<HealthCheck> getReadinessCheck() throws TaskException {
```

Comment:
Yes. It had been here since both the executor and scheduler had been reading it. The scheduler cases were somewhat special and I was able to move them into the Scheduler-specific wrappers. This has now been moved into ExecutorLabelReader.